contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -411,6 +459,26 @@ contract AerodromeAMOStrategy is InitializableAbstractStrategy {
         _rebalance(_amountToSwap, _swapWeth, _minTokenReceived);
     }
 
+    function rebalanceWithVolumeLimit(
+        uint256 _amountToSwap,
+        bool _swapWeth,
+        uint256 _minTokenReceived
+    ) external nonReentrant onlyRebalancer {
+        uint256 volumeRebalancedCurrentPeriod = _amountToSwap;
+
+        // no rebalance transaction from the rebalancer in the last hour
+        if (rebalancerVolumeTally.startBlock + REBALANCER_TRADE_INTERVAL_BLOCKS < block.number) {
+            rebalancerVolumeTally.startBlock = block.number;
+        } else {
+            volumeRebalancedCurrentPeriod += rebalancerVolumeTally.volumeRebalaced;
+        }
+
+        require(volumeRebalancedCurrentPeriod <= rebalancerVolumeLimit, "Amount to swap exceeds limit");
+        rebalancerVolumeTally.volumeRebalaced = volumeRebalancedCurrentPeriod;
+
+        _rebalance(_amountToSwap, _swapWeth, _minTokenReceived);
```

Comment:
Would be great to ensure that minTokenReceived isn't some crazy low amount

Reply:
good point I guess we could enforce it to be 1bp below swapping amount. Since the automatic rebalancing shouldn't work outside the [-1,0] tick. And by work I mean start and end point being the tick